compose/config/fields_schema.json

```diff
@@ -37,7 +37,15 @@
 
         "environment": {
```

Comment:
Oh, hmm, so I guess dashes are valid in label and hostnames, so that is one issue. We could just relax this to "any character" which would make it usable for all three.